tests/test_core.py

```diff
@@ -1915,8 +1916,8 @@ def test_em_asm_arguments_side_effects(self):
     'linked': (['-s', 'MAIN_MODULE'],),
   })
   def test_em_js(self, args):
-    if 'MAIN_MODULE' in args and self.get_setting('ALLOW_MEMORY_GROWTH') and not self.is_wasm():
-      self.skipTest('main module not compatible with asm.js memory growth')
+    if 'MAIN_MODULE' in args and self.get_setting('WASM') == 0:
```

Comment:
This looks odd.  Why not `!is_wasm()`?

Reply:
`is_wasm()` does not detect wasm2js, which may be confusing, yeah. The definition is
```
  def is_wasm(self):
    return self.is_wasm_backend() or self.get_setting('WASM') != 0
```
So when the wasm backend is true, it always returns true. Perhaps we should change that, I'll investigate.